streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -763,7 +763,7 @@ public boolean checkStateUpdater(final long now,
         if (stateUpdater.restoresActiveTasks()) {
             handleRestoredTasksFromStateUpdater(now, offsetResetter);
         }
-        return !stateUpdater.restoresActiveTasks();
+        return !stateUpdater.restoresActiveTasks() && !tasks.pendingTasksToRecycleExist();
```

Comment:
I am not familiar with this code. The method name is `checkStateUpdater` -- but check for what? Semantics are not clear to me. Could we find a better name or add a comment?

Reply:
I will consider to rename this method in a different PR since I am thinking about to split method in three methods -- one for adding tasks to the state updater, one for getting tasks from the state updater, and one for checking whether the stream thread can change state to RUNNING.